lamindb_setup/dev/upath.py

```diff
@@ -394,43 +391,44 @@ def convert_pathlike(pathlike: Union[str, Path, UPath]) -> UPath:
     return path
 
 
-def create_path(path: UPath) -> UPath:
-    global AWS_CREDENTIALS_PRESENT
+hosted_buckets = (
+    "s3://lamin-eu-central-1",
+    "s3://lamin-us-west-2",
+)
+credentials_cache: Dict[str, Dict[str, str]] = {}
 
+
+def create_path(path: UPath) -> UPath:
     path = convert_pathlike(path)
+    # test whether we have an AWS S3 path
     if not isinstance(path, S3Path):
         return path
-    path = UPath(path, cache_regions=True)
-    # the below is problematic, because it will assume that all subsequent
-    # requests will then be treated as anon, and we'll never try something else
-    # if isinstance(path, S3Path) and not upath.AWS_CREDENTIALS_PRESENT:
-    #     # if we already know we don't have AWS credentials
-    #     # let's make an anon request from the get-go
-    #     return UPath(path, anon=True)
-    try:
-        path.fs.call_s3("head_bucket", Bucket=path._url.netloc)
-        AWS_CREDENTIALS_PRESENT = True  # type: ignore
-        return path
-    except NoCredentialsError:
-        # we don't have any credentials at all
-        # for legacy reasons, we store information that no credentials
-        # are present
-        AWS_CREDENTIALS_PRESENT = False  # type: ignore
-        return UPath(path, anon=True)
-    except Exception:
-        # we have credentials but they're wrong
-        # either,
-        # we try to access a public bucket in a different AWS account
-        # or we have the credentials for the wrong AWS acount
-        # or we have expired credentials
-        from ._hub_core import access_aws
-
-        access_aws()
-        # the env variables aren't recognized if previously
-        # another .fs.session._credentials object was already created
+    # test whether AWS credentials are present
+    session = botocore.session.get_session()
+    credentials = session.get_credentials()
+    if credentials is None or credentials.access_key is None:
+        anon = True
+    else:
+        anon = False
+    # test whether we are on hosted storage or not
+    path_str = path.as_posix()
+    is_hosted_storage = path_str.startswith(hosted_buckets)
+
+    if not is_hosted_storage:
+        # make anon request if no credentials present
+        return UPath(path, cache_regions=True, anon=anon)
+    else:
+        root_folder = path_str.replace("s3://", "").split("/")[0]
```

Comment:
Again this happens on every call of create_path.

Reply:
Yes! Calling you!